src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -619,7 +619,8 @@ void SpatialPooler::compute(UInt inputArray[], bool learn,
   if (learn) {
     boostOverlaps_(overlaps_, boostedOverlaps_);
   } else {
-    boostedOverlaps_.assign(overlaps_.begin(), overlaps_.end());
+    for (auto overlap : overlaps_)
+      boostedOverlaps_.push_back(overlap);
```

Comment:
Why doesn't `assign` work? Alternatively, can we use the vector copy constructor:

```
boostedOverlaps_ = overlaps_;
```

The current method doesn't clear `boostedOverlaps_` before which could be a bug. But also the one liners are cleaner and might be more efficient.